clients/src/test/java/org/apache/kafka/test/TestUtils.java

```diff
@@ -542,28 +542,41 @@ public static Set<TopicPartition> generateRandomTopicPartitions(int numTopic, in
     }
 
     /**
-     * Assert that a future raises an expected exception cause type. Return the exception cause
-     * if the assertion succeeds; otherwise raise AssertionError.
+     * Assert that a future raises an expected exception cause type and optionally verify the error message.
```

Comment:
Why are you rewriting the two methods, it's OK to keep it unchanged if no specific purpose are intended.

Reply:
@dengziming you're right, reverted. Thanks!